src/video/win32_v.cpp

```diff
@@ -1085,45 +1085,43 @@ static const Dimension default_resolutions[] = {
 
 static void FindResolutions()
 {
-	uint n = 0;
 	uint i;
 	DEVMODEA dm;
 
 	/* Check modes for the relevant fullscreen bpp */
 	uint bpp = _support8bpp != S8BPP_HARDWARE ? 32 : BlitterFactory::GetCurrentBlitter()->GetScreenDepth();
 
+	_resolutions.clear();
+
 	/* XXX - EnumDisplaySettingsW crashes with unicows.dll on Windows95
 	 * Doesn't really matter since we don't pass a string anyways, but still
 	 * a letdown */
 	for (i = 0; EnumDisplaySettingsA(nullptr, i, &dm) != 0; i++) {
-		if (dm.dmBitsPerPel == bpp &&
-				dm.dmPelsWidth >= 640 && dm.dmPelsHeight >= 480) {
+		if (dm.dmBitsPerPel == bpp && dm.dmPelsWidth >= 640 && dm.dmPelsHeight >= 480) {
 			uint j;
 
-			for (j = 0; j < n; j++) {
+			for (j = 0; j < _resolutions.size(); j++) {
 				if (_resolutions[j].width == dm.dmPelsWidth && _resolutions[j].height == dm.dmPelsHeight) break;
 			}
 
 			/* In the previous loop we have checked already existing/added resolutions if
 			 * they are the same as the new ones. If this is not the case (j == n); we have
 			 * looped all and found none, add the new one to the list. If we have reached the
 			 * maximum amount of resolutions, then quit querying the display */
-			if (j == n) {
+			if (j == _resolutions.size()) {
+				_resolutions.push_back(Dimension());
 				_resolutions[j].width  = dm.dmPelsWidth;
 				_resolutions[j].height = dm.dmPelsHeight;
-				if (++n == lengthof(_resolutions)) break;
 			}
 		}
 	}
 
 	/* We have found no resolutions, show the default list */
-	if (n == 0) {
-		memcpy(_resolutions, default_resolutions, sizeof(default_resolutions));
-		n = lengthof(default_resolutions);
+	if (_resolutions.size() == 0) {
+		for (uint n = 0; n < lengthof(default_resolutions); n++) _resolutions.push_back(default_resolutions[n]);
 	}
```

Comment:
perhaps `_resolutions = default_resolutions;` ? I think everything would be copied properly...